src/cc_helpers.cc

```diff
@@ -1167,49 +1168,32 @@ void print_org_name(X509_NAME* name) {
   printf("\n");
 }
 
-void print_ssl_error(int code) {
-  switch(code) {
-  case SSL_ERROR_NONE:
-    printf("No ssl error\n");
-    break;
-  case SSL_ERROR_WANT_READ:
-    printf("want read ssl error\n");
-    break;
-  case SSL_ERROR_WANT_WRITE:
-    printf("want write ssl error\n");
-    break;
-#ifndef BORING_SSL
-  case SSL_ERROR_WANT_CONNECT:
-    printf("want connect ssl error\n");
-    break;
-#endif
-  case SSL_ERROR_WANT_ACCEPT:
-    printf("want accept ssl error\n");
-    break;
-  case SSL_ERROR_WANT_X509_LOOKUP:
-    printf("want lookup ssl error\n");
-    break;
+/* Declare an array of SSL Error codes mapped to an error description */
+optlookup Ssl_errors[] =
+  {
+      DCL_OPTLOOKUP(SSL_ERROR_NONE                 , "No ssl error")
+    , DCL_OPTLOOKUP(SSL_ERROR_WANT_READ            , "want read error")
+    , DCL_OPTLOOKUP(SSL_ERROR_WANT_WRITE           , "want write error")
```

Comment:
The `DCL_OPTLOOKUP()` macros will define the decription to be, e.g,  `"SSL_ERROR_WANT_WRITE: want write error"`.

I, intentionally, wanted the mnemonic also to be reported in the description so that one can grep for it in code.

Do we want the string `want write error` also? 

I think, and prefer, that just the mnemonic in " "; i.e. `"SSL_ERROR_WANT_WRITE"` should be sufficiently clear.

I just added the trailing text `want write error` as that's what the old code was doing.

I'm ok with this extra text ... but it is really superfluous (in my opinion).

Can go either way.

Reply:
It's fine the way you did it.